wire-ios-sync-engine/Source/Use cases/DisableAnalyticsUseCase.swift

```diff
@@ -29,21 +29,34 @@ public protocol DisableAnalyticsUseCaseProtocol {
 
 /// Concrete implementation of the DisableAnalyticsUseCaseProtocol.
 /// This struct is responsible for disabling analytics sharing.
-public struct DisableAnalyticsUseCase: DisableAnalyticsUseCaseProtocol {
+struct DisableAnalyticsUseCase: DisableAnalyticsUseCaseProtocol {
 
-    private let analyticsManager: AnalyticsManagerProtocol
+    private let sessionManager: AnalyticsManagerProviding
+    private let analyticsSessionProvider: DisableAnalyticsUseCaseAnalyticsSessionProviding
 
-    /// Initializes a new instance of DisableAnalyticsUseCase.
+    /// Initializes a new instance of `DisableAnalyticsUseCase`.
     ///
-    /// - Parameter analyticsManager: The analytics manager to use for disabling tracking.
-    public init(analyticsManager: AnalyticsManagerProtocol) {
-        self.analyticsManager = analyticsManager
+    /// - Parameters:
+    ///   - sessionManager: The session manager that conforms to `AnalyticsManagerProviding` for managing analytics sessions.
+    ///   - analyticsSessionProvider: An instance conforming to `DisableAnalyticsUseCaseAnalyticsSessionProviding` that manages the user's analytics session state.
+    init(
+        sessionManager: AnalyticsManagerProviding,
+        analyticsSessionProvider: DisableAnalyticsUseCaseAnalyticsSessionProviding
+    ) {
+        self.sessionManager = sessionManager
+        self.analyticsSessionProvider = analyticsSessionProvider
     }
 
     /// Invokes the use case to disable analytics sharing.
     ///
     /// This method calls the `disableTracking` method on the analytics manager if it exists.
-    public func invoke() {
-        analyticsManager.disableTracking()
```

Comment:
Why wasn't this enough before? It seems like a bug that even if this is called, that the events are still being sent, regardless of the `AnalyticsManager` existing.

Reply:
@agisilaos ?

Reply:
@johnxnguyen Not really sure to be honest. As soon as we took the analytics manager out of the equation we stopped receiving events when disabled. After we merge everything to the feature branch. I want to take another look at the whole refactor and set of changes as a good practice to see what we can improve. 

For now is it ok to go ahead and merge it?

Reply:
Your solution has introduce some complexity that would be nice to avoid. Since this is somewhat urgent, I'm happy to approve it but I think the time should be taken to get to the bottom of the issue.

Reply:
But then... if we know that `disableAnayltics()` doesn't work, why do we still call it?

Reply:
Sounds good! 👍

Reply:
We're going to revisit with @johnxnguyen after everything is merged.